upf/upf_pfcp_api.c

```diff
@@ -96,9 +96,18 @@ upf_pfcp_api_session_data_init (void *sxp, time_t start_time)
 }
 
 static void
-init_response_node_id (struct pfcp_response *r)
+init_response_node_id (struct pfcp_response *response)
 {
-  //TODO: need CLI/API to set local Node-Id.....
+  upf_main_t *gtm = &upf_main;
+  if (gtm->node_id.type == NID_FQDN)
+    {
+      vec_reset_length (response->node_id.fqdn);
```

Comment:
that construct could leak memory, you need to free `response->node_id.fqdn` if you expect it contain a odl value.

**But** can `response->node_id.fqdn` really contain something valid at this point? It should be sufficient to plainly initialize the field (see next comment).

Reply:
Fixed
Thanks